lua/luasnip/session/snippet_collection.lua

```diff
@@ -172,8 +172,10 @@ function M.clean_invalidated(opts)
 
 	-- remove invalidated snippets from all tables.
 	for _, type_snippets in pairs(by_prio) do
-		for _, prio_snippets in pairs(type_snippets) do
-			prio_snippets = without_invalidated(prio_snippets)
+		for priority, prio_snippets in pairs(type_snippets) do
+			if priority ~= 'order' then
+				prio_snippets = without_invalidated(prio_snippets)
```

Comment:
This should also be `by_prio[priority] = ...`.
Imo key instead of priority would be a bit better, what do you think?

Reply:
that actually sounds better

Reply:
My bad, I didn't look properly, it should be `type_snippets[key] = ...` ofc :facepalm: